File: packages/urdu-js/tests/string-utils/count.test.ts
import { describe, it, expect } from 'vitest'
import { wordCount, charCount } from '../../src/string-utils/count.js'

describe('wordCount', () => {
  it('empty → 0', () => expect(wordCount('')).toBe(0))
  it('one word → 1', () => expect(wordCount('کتاب')).toBe(1))
  it('two words → 2', () => expect(wordCount('کتاب ہے')).toBe(2))
  it('whitespace only → 0', () => expect(wordCount('   ')).toBe(0))
})

describe('charCount', () => {
  it('empty → 0', () => expect(charCount('')).toBe(0))
  it('basic count', () => expect(charCount('کتاب')).toBe(4))
  it('excludeDiacritics', () => expect(charCount('عِلم', { excludeDiacritics: true })).toBe(3))
  it('excludeWhitespace', () => expect(charCount('ک ت', { excludeWhitespace: true })).toBe(2))
})

File: packages/urdu-dart/test/string_utils/count_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('wordCount', () {
    test('empty -> 0', () => expect(wordCount(''), equals(0)));
    test('one word -> 1', () => expect(wordCount('کتاب'), equals(1)));
    test('two words -> 2', () => expect(wordCount('کتاب میں'), equals(2)));
    test('whitespace only -> 0', () => expect(wordCount('   '), equals(0)));
  });

  group('charCount', () {
    test('empty -> 0', () => expect(charCount(''), equals(0)));
    test('basic count', () => expect(charCount('کتاب'), equals(4)));
  });
}
